duckduckgo/Android
# Revert keeping HTTP auth credentials when site is fireproof(ed) #1066

Task/Issue URL: https://app.asana.com/0/488551667048375/1199707485317446/f
Tech Design URL: 
CC: 

**Description**:
The HTTP auth DB can caused some crashes for API<26 in some OEMs due to the DB being locked in some scenarios when firing.
This PR:
- removes the HTTP auth DB we introduced to selectively delete HTTP auth credentials. Back to just using `WebViewDatabase`
- keeps some tidy up that was done back then


**Steps to test this PR**:
1. Install 5.76.0
2. go to httpbin.org/basic-auth/u/p
3. user = u, pass = p
4. Upgrade the app to the one in this branch
5. go to httpbin.org/basic-auth/u/p
6. verify login is NOT required
7. tap on fire button
8. go to httpbin.org/basic-auth/u/p
9. verify IS required
10. go to httpbin.org/basic-auth/u/p
11. user = u, pass = p
12. fireproof site
13. tap on fire button
14. go to httpbin.org/basic-auth/u/p
15. verify login IS required

(repeat test above for initial install of app v 5.75.x)


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
